src/Swarm/Game/Step.hs

```diff
@@ -1243,14 +1243,21 @@ execConst c vs s k = do
       let Location x y = loc
       return $ Out (VPair (VInt (fromIntegral x)) (VInt (fromIntegral y))) s k
     Detect -> case vs of
-      [VText name, VPair (VPair (VInt x1) (VInt y1)) (VPair (VInt x2) (VInt y2))] -> do
+      [VText name, r] -> do
```

Comment:
There is a subtle issue here, which is that this pattern now matches *any* `Value` as the second argument to `Detect` instead of only pairs of pairs of ints.  In theory, if typechecking works, that is the only possible value we could ever get here. But in case there is ever any kind of bug that causes the wrong kind of `Value` to end up here, previously it would have fallen through to the `badConst` case, which throws an appropriate in-game exception; with the current code, it will instead actually *crash* the whole game with a non-exhaustive pattern match error in the `rectCells` function.

We should be using a `VRect` pattern *here*, in the pattern for `vs`, and then pass the resulting four `Int` values to `rectCells`.  That way `rectCells` will be a total function.